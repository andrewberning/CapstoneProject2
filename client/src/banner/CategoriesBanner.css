.CategoriesBanner {
  background-color: salmon;
  text-align: center;
}

.categories-title-container {
  padding-top: 20px;
}

.categories-title-container h2 {
  margin: 0;
}

.all-categories-link-container {
  padding: 20px;
}

.all-categories-link-container a {
  text-decoration: none;
  color: white;
}