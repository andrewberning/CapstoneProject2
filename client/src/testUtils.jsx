import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
