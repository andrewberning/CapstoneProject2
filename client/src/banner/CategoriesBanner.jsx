import { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import CategoriesCardList from "../categories/CategoriesCardList";
import ShoplyApi from '../api/api';
import "./CategoriesBanner.css";

export default function CategoriesBanner() {
  const [categories, setCategories] = useState([]);

  useEffect(function getFiveCategoriesOnMount() {
    async function getCategories(category) {
      let categories = await ShoplyApi.getCategories(category);
      setCategories(categories);
    }

    getCategories();
  }, [])

  return (
    <section className="CategoriesBanner">
        <div className="category-title text-center">
          <h2 className="mb-4 font-weight-bold">Featured Categories</h2>
          <Link to={"/categories"}>See all</Link>
        </div>
        <CategoriesCardList categories={categories.slice(0,5)} />

    </section>
  );
}