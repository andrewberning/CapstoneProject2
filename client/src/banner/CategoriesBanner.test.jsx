import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CategoriesBanner from "./CategoriesBanner";
import { UserProvider } from "../testUtils";

describe('CategoriesBanner component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <CategoriesBanner />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <CategoriesBanner />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});