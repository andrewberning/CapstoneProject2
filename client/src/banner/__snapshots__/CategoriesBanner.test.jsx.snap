// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoriesBanner component > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="CategoriesBanner"
  >
    <div
      class="categories-title-container"
    >
      <h2
        class="categories-title"
      >
        Featured Categories
      </h2>
    </div>
    <div
      class="all-categories-link-container"
    >
      <button
        class="btn btn-primary"
      >
        <a
          class="all-categories-link"
          href="/categories"
        >
          See all categories
        </a>
      </button>
    </div>
    <div
      class="CategoriesBannerList"
    >
      <ul
        class="categories-list"
      />
    </div>
  </section>
</DocumentFragment>
`;
